airbyte-integrations/connectors/source-github/source_github/streams.py

```diff
@@ -743,21 +702,33 @@ def path(
     ) -> str:
         return "graphql"
 
+    def raise_error_from_response(self, response_json):
+        if "errors" in response_json:
+            raise Exception(str(response_json["errors"]))
+
+    def _get_name(self, repository):
+        return repository["owner"]["login"] + "/" + repository["name"]
+
     def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
-        nodes = response.json()["data"]["repository"]["pullRequests"]["nodes"]
-        for record in nodes:
-            record["review_comments"] = sum([node["comments"]["totalCount"] for node in record["review_comments"]["nodes"]])
-            record["comments"] = record["comments"]["totalCount"]
-            record["commits"] = record["commits"]["totalCount"]
-            record["repository"] = record["repository"]["name"]
-            if record["merged_by"]:
-                record["merged_by"]["type"] = record["merged_by"].pop("__typename")
-            yield record
+        self.raise_error_from_response(response_json=response.json())
+        repository = response.json()["data"]["repository"]
+        if repository:
```

Comment:
is this check because repository can be absent from `response.json()["data"]` or because it can be an empty object?

Reply:
If we query in GraphQL non existent repository, the response will be:
```
{'data': {'repository': None},
 'errors': [{'locations': [{'column': 3, 'line': 2}],
             'message': "Could not resolve to a Repository with the name 'airbytehq/airbytea'.",
             'path': ['repository'],
             'type': 'NOT_FOUND'}]}
```

but yes, it will be catched by `raise_error_from_response` because it's also error `NOT_FOUND`